cookbook/core/control_flow/map_task.py

```diff
@@ -65,5 +65,78 @@ def my_map_workflow(a: typing.List[int]) -> str:
     print(f"{result}")
 
 # %%
-# By default, the map task uses the K8s Array plugin. Map tasks can also run on alternate execution backends, such as `AWS Batch <https://docs.flyte.org/en/latest/deployment/plugin_setup/aws/batch.html#deployment-plugin-setup-aws-array>`__,
+# When defining a map task, avoid calling other tasks in it. Flyte
+# can't accurately register tasks that call other tasks.  While Flyte
+# will correctly execute a task that calls other tasks, it will not be
+# able to give the full performance advantages. This is
+# especially true for map tasks.
+#
+# In this example, the map task ``suboptimal_mappable_task`` would not
+# give you the best performance.
+@task
+def upperhalf(a: int) -> int:
+    return a / 2 + 1
+
+@task
+def suboptimal__mappable_task(a: int) -> str:
+    inc = upperhalf(a=a)
+    stringified = str(inc)
+    return stringified
```

Comment:
Why do we need to highlight this considering that a task within a task isn't proposed by Flyte in a general sense?

Reply:
@samhita-alla Good question. 

Is there a place in the doc where we are explicit about the consequences of calling tasks within tasks? If so, I agree that I'm being redundant here.

My rationale (from a point brought up by @eapolinario ): calling a task within a task isn't handled by registration. Most notably for map tasks, this reduces opportunities for parallel processing.

Does that make sense?

Reply:
> Is there a place in the doc where we are explicit about the consequences of calling tasks within tasks? If so, I agree that I'm being redundant here.

I don't think we're highlighting this point anywhere in the docs. However, since we highlight the fact that the task is nothing more than a Python function, writing one within it doesn't mean anything new to Flyte. It doesn't lead to any consequences as such; it's more of -- the task is not part of the DAG anymore and Flyte only recognizes the parent task.

> calling a task within a task isn't handled by registration. Most notably for map tasks, this reduces opportunities for parallel processing.
This makes sense, however.